File: code/test_suites/cpp/FunctionBuildTest/src/function_pyphonetics_phonetics.cpp
#include <string>
#include <soundex.h>

class SoundexProcessor {
public:
    std::string generate_soundex(const std::string& input_string) {
        if (input_string.empty()) {
            return "The given string is empty.";
        }
        Soundex soundex;
        return soundex.encode(input_string);
    }
};

File: manual/code/generate_strategies/IR(summary)/java/java/function_pyphonetics_phonetics.java
package com.example;

import org.apache.commons.codec.language.Soundex;

public class SoundexProcessor {
    public String generate_soundex(String input_string) {
        class SoundexProcessor {
            public String generate_soundex(String input_string) {
                if (input_string.equals("")) {
                    return "The given string is empty.";
                }
                Soundex soundex = new Soundex();
                return soundex.encode(input_string);
            }
        }
        return new SoundexProcessor().generate_soundex(input_string);
    }
}